File: 217. Contains Duplicate/python/217_contains_duplicate.py
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        check = False   
        num_map = {}
        for n in nums:
            if n not in num_map:
                num_map[n] = True
            else:
                check = True
        return check
        
        # Other possible solution
        # return len(nums) != len(set(nums))
    
sol = Solution()

test1 = [1,2,3,1]
test2 = [1,2,3,4]
test3 = [1,1,1,3,3,4,3,2,4,2]


print(sol.containsDuplicate(test1))
print(sol.containsDuplicate(test2))
print(sol.containsDuplicate(test3))

File: 217. Contains Duplicate/go/217_contains_duplicate.go
package main

import "fmt"

func main() {
	test1 := []int{1, 2, 3, 1}
	test2 := []int{1, 2, 3, 4}
	test3 := []int{1, 1, 1, 3, 3, 4, 3, 2, 4, 2}

	fmt.Println(containsDuplicate(test1))
	fmt.Println(containsDuplicate(test2))
	fmt.Println(containsDuplicate(test3))
}

func containsDuplicate(nums []int) bool {
	mymap := make(map[int]int)

	for _, n := range nums {
		mymap[n]++
		if mymap[n] > 1 {
			return true
		}
	}

	return false
}
